rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/CloudProviderBakeHandler.groovy

```diff
@@ -212,14 +218,26 @@ abstract class CloudProviderBakeHandler {
       }
     }
 
-    def finalTemplateFileName = bakeRequest.template_file_name ?: getTemplateFileName(selectedOptions.baseImage)
-    def finaltemplateFilePath = "$configDir/$finalTemplateFileName"
+    String finalTemplateFileName
+    String finalTemplateFilePath
+
+    if (bakeRequest.template) {
+        def tempFile = Files.createTempFile("template_${bakeRequest.request_id}_", ".json")
+        FileCopyUtils.copy(bakeRequest.template.getBytes(StandardCharsets.UTF_8), tempFile.toFile())
+
+        finalTemplateFileName = tempFile.fileName
+        finalTemplateFilePath = tempFile
```

Comment:
Since we are pretty far away from the exit of the packer command in this bit of code, I added a line to have the file deleted on VM exit. I think that that should be enough, but let me know if not because it would take some larger reworking to delete immediately on completion of the bake.

Reply:
One option to consider is similar to what I did in https://github.com/spinnaker/rosco/pull/253/files, where I delete the temporary file in `completeBake` which is a callback upon completion of the packer command.